### PR TITLE
webfilter getAttributeNames fix

### LINKED ISSUE #7033: HazelcastHttpSession getAttributeNames returns entries that were only read but not written
Reopening https://github.com/hazelcast/hazelcast/issues/6868.
HazelcastHttpSession::getAttributeNames returns entries (with null values) if HazelcastHttpSession::getAttribute was called with those entry names.
So, in the example below, we call hazelcastHttpSession.getAttribute("test") it's being returned when calling hazelcastHttpSession.getAttributeNames().

This is the test code that reproduces it ("test" is printed even though it was only read, not write):
				
				package test;
				
				import java.util.Enumeration;
				
				import javax.servlet.FilterConfig;
				import javax.servlet.ServletContext;
				import javax.servlet.ServletException;
				
				import org.eclipse.jetty.servlet.ServletContextHandler;
				import org.eclipse.jetty.servlet.ServletContextHandler.Context;
				
				import com.hazelcast.web.HazelcastHttpSession;
				import com.hazelcast.web.WebFilter;
				
				public class HazelcastSessionParamCheck {
					public static void main(String[] args) throws ServletException {
						
						WebFilter filter = new WebFilter();
						final Context servletContext = new ServletContextHandler().new Context(); 
						
						FilterConfig filterConfig = new FilterConfig() {
							
							public ServletContext getServletContext() {
								// TODO Auto-generated method stub
								return servletContext;
							}
							
							public Enumeration<String> getInitParameterNames() {
								// TODO Auto-generated method stub
								return null;
							}
							
							public String getInitParameter(String name) {
								// TODO Auto-generated method stub
								return null;
							}
							
							public String getFilterName() {
								// TODO Auto-generated method stub
								return null;
							}
						};
						filter.init(filterConfig);
						HazelcastHttpSession hazelcastHttpSession = new HazelcastHttpSession(filter, "sessionId", null, true, true); 
						hazelcastHttpSession.getAttribute("test");
						Enumeration<String> sessionAttributeNames = hazelcastHttpSession.getAttributeNames();
						while (sessionAttributeNames.hasMoreElements()) {
							System.out.println(sessionAttributeNames.nextElement());
						}
						System.exit(1);
						
					}
				}

With this maven configuration:
				  
				<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
				  <modelVersion>4.0.0</modelVersion>
				  <groupId>test</groupId>
				  <artifactId>test</artifactId>
				  <version>0.0.1-SNAPSHOT</version>
				  <name>test</name>
				  
				  <dependencies>
				  	<dependency>
						<groupId>com.hazelcast</groupId>
						<artifactId>hazelcast-wm</artifactId>
						<version>3.5.4</version>
					</dependency>
					
					<dependency>
						<groupId>javax.servlet</groupId>
						<artifactId>javax.servlet-api</artifactId>
						<version>3.0.1</version>
					</dependency>
					
					<dependency>
						<groupId>org.eclipse.jetty</groupId>
						<artifactId>jetty-webapp</artifactId>
						<version>8.1.18.v20150929</version>
					</dependency>
				  </dependencies>
				</project>

### PR DESCRIPTION
**Pull request by bilalyasar**
fixes #7033 , you can find detailed information at #7033

### CODE REVIEW

**Comment by mesutcelik** [comment]
checkstyle problems...

**Comment by bilalyasar** [comment]
checkstyle is fixed

File: checkstyle/suppressions.xml [hunk 1]
```diff
@@ -237,6 +237,7 @@
     <suppress checks="JavadocType" files="com/hazelcast/web/"/>
     <suppress checks="JavadocMethod" files="com/hazelcast/web/"/>
     <suppress checks="JavadocVariable" files="com/hazelcast/web/"/>
+    <suppress checks="NPathComplexity" files="com/hazelcast/web/LocalCacheEntry.java"/>
```
Comment (mesutcelik) [inline checkstyle/suppressions.xml]: can't you fix this? It feels `equals` method is adding this complexity..

**Comment by bilalyasar** [comment]
updated

**Comment by mesutcelik** [comment]
:+1:  can you please forward port?

**Pull request merged**